From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Subject: [PATCH] ttyname_r returning EINVAL

2011-08-05  Samuel Thibault  <samuel.thibault@ens-lyon.org>

        * sysdeps/mach/hurd/ttyname_r.c (__ttyname_r): Return ERANGE instead of
        EINVAL when BUFLEN is too smal.

---
http://sourceware.org/ml/libc-alpha/2011-08/msg00013.html

 sysdeps/mach/hurd/ttyname_r.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sysdeps/mach/hurd/ttyname_r.c b/sysdeps/mach/hurd/ttyname_r.c
index 5f6c9c3..5718fca 100644
--- a/sysdeps/mach/hurd/ttyname_r.c
+++ b/sysdeps/mach/hurd/ttyname_r.c
@@ -43,7 +43,7 @@ __ttyname_r (int fd, char *buf, size_t buflen)
   len = strlen (nodename) + 1;
   if (len > buflen)
     {
-      errno = EINVAL;
+      errno = ERANGE;
       return errno;
     }
 
-- 
tg: (0234227..) t/ttyname_ERANGE (depends on: baseline)
